### DISCUSSION TITLE
Feature recommendation: fetching all paths during getStaticPaths (of pages, of posts, and of custom post types).

### DISCUSSION THREAD

**Discussion by roeean** [wpengine/faustjs#967] [Ideas]
Hi,

My project has many pages and posts - most of them are "custom-type" posts. In Faust.js documentation, you showed an example of how to pull the "first five pages" while I created a function that I think might interest you:

To guarantee the revalidation of all pages and posts, the function is pulling all their paths in getStaticPaths:

```
export const getStaticPaths = async () => {
  const { query, inlineResolved } = client.client;

  //* Fetch all post types that exist in graphQL
  const postTypes = await inlineResolved(() => [...query.contentTypes().nodes]
    .map(({ graphqlPluralName }) => graphqlPluralName)
    .filter(type => type !== 'mediaItems')
  );

  //* Fetch all paths
  const paths = await inlineResolved(() => postTypes.reduce(
    (prev, cur) => {
      const typeFn = cur[0].toLowerCase() + cur.substring(1);
      return [...prev, ...query[typeFn]()?.nodes.map(({ uri }) => uri)];
    }, []));

  return {
    fallback: 'blocking',
    paths
  };
};
```

Maybe consider adding such an option in one of the following versions,

Roee Angel

**Comment by astratagem** [comment]
I agree that this would be a pretty great addition. 

Worth noting that the upcoming app directory support will support a new approach to this:

https://beta.nextjs.org/docs/data-fetching/generating-static-params

Also, for reference, here are the docs for using the existing support for getStaticPaths:

https://nextjs.org/docs/basic-features/data-fetching/get-static-paths
